### PR TITLE
r.flowaccumulation: Add a new addon that implements the Memory-Efficient Flow Accumulation parallel algorithm (Cho 2023)

### PR DESCRIPTION
**Pull request by HuidaeCho**
This PR adds a new addon for fast parallel flow accumulation using OpenMP. It implements https://doi.org/10.1016/j.envsoft.2023.105771. Ported from https://github.com/HuidaeCho/mefa.

Benchmark against r.accumulate (28.216s vs. 50.181s) for 30m DEM for Texas on i7-1370P CPU with 64GB RAM
image

```
$ time r.accumulate direction=drain accumulation=accum_old
Input direction format assumed to be degrees CCW from East divided by 45
Reading direction map...
 100%
Allocating buffers...
 100%
Accumulating flows iteratively...
 100%
Writing accumulation map...
 100%

real    0m50.181s
user    0m46.074s
sys     0m3.938s

$ time r.flowaccumulation input=fdr output=accum_new 
Using 20 threads for parallel computation
Reading flow direction raster <fdr>...
Flow direction format assumed to be powers of 2 CW from East
 100%
Input time for flow direction: 6.835615 seconds
Accumulating flows...
Compute time for flow accumulation: 3.490350 seconds
Writing flow accumulation raster <accum_new>...
 100%
Output time for flow accumulation: 17.872626 seconds
Total elapsed time: 28.198680 seconds

real    0m28.216s
user    1m14.207s
sys     0m6.489s
```

### CODE REVIEW

File: src/raster/r.flowaccumulation/r.flowaccumulation.html [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+<p>For comparison, these numbers show the new flow accumulation by
+<em>r.flowaccumulation</em> (<i>flow_accum_new</i>). The same cells are
+properly accumulated from the headwater cells.
+
+<p><img src="r_flowaccumulation_nc_comparison.png">
+
+<h2>SEE ALSO</h2>
+
+<em>
+<a href="https://grass.osgeo.org/grass-stable/manuals/addons/r.accumulate.html">r.accumulate</a>,
```
Comment (neteler) [inline src/raster/r.flowaccumulation/r.flowaccumulation.html]:
```suggestion
<a href="r.accumulate.html">r.accumulate</a>,
```

For addons, please just use the relative path.

File: src/raster/r.flowaccumulation/r.flowaccumulation.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+<a href="https://grass.osgeo.org/grass-stable/manuals/r.watershed.html">r.watershed</a>,
+<a href="https://grass.osgeo.org/grass-stable/manuals/r.stream.extract.html">r.stream.extract</a>,
+<a href="r.stream.distance.html">r.stream.distance</a>
+</em>
+
+<h2>REFERENCES</h2>
+
+Huidae Cho, July 2023. <em>Memory-Efficient Flow Accumulation Using a Look-Around Approach and Its OpenMP Parallelization</em> Environmental Modelling &amp; Software 167, 105771. <a href="https://doi.org/10.1016/j.envsoft.2023.105771">10.1016/j.envsoft.2023.105771</a>.
```
Comment (neteler) [inline src/raster/r.flowaccumulation/r.flowaccumulation.html]:
```suggestion
Huidae Cho, July 2023. <em>Memory-Efficient Flow Accumulation Using a Look-Around Approach and
Its OpenMP Parallelization</em> Environmental Modelling &amp; Software 167, 105771.
<a href="https://doi.org/10.1016/j.envsoft.2023.105771">10.1016/j.envsoft.2023.105771</a>.
```

File: src/raster/r.flowaccumulation/r.flowaccumulation.html (hunk 1 above)
Comment (HuidaeCho) [reply src/raster/r.flowaccumulation/r.flowaccumulation.html]: @neteler why do we use relative paths only for addons?

Comment (neteler) [reply src/raster/r.flowaccumulation/r.flowaccumulation.html]: On the main server they will work, but also on the mirror sites.
With hardcoded path we would always point to the main site, weakening the idea of the mirrors.

Comment (HuidaeCho) [reply src/raster/r.flowaccumulation/r.flowaccumulation.html]: But we cannot do that for core modules for any reason?

**Pull request merged**